core/cli/src/rc-file.ts

```diff
@@ -1,19 +1,41 @@
-import { cosmiconfig } from 'cosmiconfig'
+import { styles as s } from '@dotcom-tool-kit/logger'
 import { RCFile } from '@dotcom-tool-kit/types/src'
+import { cosmiconfig } from 'cosmiconfig'
+import * as path from 'path'
+import type { Logger } from 'winston'
 
 export const explorer = cosmiconfig('toolkit', { ignoreEmptySearchPlaces: false })
+const emptyConfig = { plugins: [], hooks: {}, options: {} }
+let rootConfig: string | undefined
 
 type RawRCFile = {
   [key in keyof RCFile]?: RCFile[key] | null
 }
 
-export async function loadToolKitRC(root: string): Promise<RCFile> {
-  const result = (await explorer.search(root)) as { config: RawRCFile | null } | null
-  if (!result || !result.config) return { plugins: [], hooks: {}, options: {} }
+export async function loadToolKitRC(logger: Logger, root: string, isAppRoot: boolean): Promise<RCFile> {
+  const result = await explorer.search(root)
+
+  if (!result?.config) {
+    return emptyConfig
+  }
+  if (isAppRoot) {
+    rootConfig = result.filepath
+  } else if (result.filepath === rootConfig) {
```

Comment:
**question:** Could `rootConfig` here not be `undefined` to make this condition true?

Reply:
`result.filepath` should never be `undefined` so that would `rootConfig` being `undefined` would never make this condition true